src/rotary-dialer-element.ts

```diff
@@ -0,0 +1,280 @@
+    digit: Digit,
+    direction: Direction = Direction.normal,
+    duration = '1000ms',
+    timingFn = 'ease'
+  ) {
+    switch (digit) {
+      case Digit.zero:
+        slots?.setAttribute('style', `animation: dialZero ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.one:
+        slots?.setAttribute('style', `animation: dialOne ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.two:
+        slots?.setAttribute('style', `animation: dialTwo ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.three:
+        slots?.setAttribute(
+          'style',
+          `animation: dialThree ${duration} 1 ${direction} ${timingFn};`
+        );
+        break;
+      case Digit.four:
+        slots?.setAttribute('style', `animation: dialFour ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.five:
+        slots?.setAttribute('style', `animation: dialFive ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.six:
+        slots?.setAttribute('style', `animation: dialSix ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.seven:
+        slots?.setAttribute(
+          'style',
+          `animation: dialSeven ${duration} 1 ${direction} ${timingFn};`
+        );
+        break;
+      case Digit.eight:
+        slots?.setAttribute(
+          'style',
+          `animation: dialEight ${duration} 1 ${direction} ${timingFn};`
+        );
+        break;
+      case Digit.nine:
+        slots?.setAttribute('style', `animation: dialNine ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      default:
+        slots?.removeAttribute('style');
+    }
+  }
+
+  private static addReverseDialerAnim(slots: SVGAElement, digit: number) {
+    // When the user leaves the mouse (eq of lifting the finger) it goes back
+    RotaryDialerElement.matchDigit(slots, digit, Direction.reverse, '2000ms');
+  }
+
+  private addDialerAnim(digit: number) {
+    this.handleRotation(digit);
+  }
+
+  private removeDialerAnim() {
+    const slots = this.shadowRoot?.querySelector('#slots');
+    slots?.classList.remove('dialer-anim');
+    slots?.removeAttribute('style');
+  }
+
+  private down(digit: number) {
+    this.dial(digit);
+  }
+
+  /**
+   * Exposed lazy dial by applying dial fn with a given arg of num from 0-9
+   * @param digit
+   */
+  public dial(digit: Digit) {
+    this.removeDialerAnim();
+    this.addDialerAnim(digit);
+    this.dispatchEvent(
+      new CustomEvent('dialer-grab', {
+        detail: { digit },
+      })
+    );
+  }
+
+  private onValueChange(event: KeyboardEvent) {
+    const target = event.target as HTMLInputElement;
+    const lastValue = target.value;
+    target.value = '';
+    const digit = parseInt(lastValue);
+    this.dial(digit);
+  }
+
+  private handleRotation(digit: number) {
+    const slots = this.shadowRoot?.querySelector('#slots') as SVGAElement;
```

Comment:
It's actually `SVGPathElement` and not [`SVGAElement`](https://developer.mozilla.org/en-US/docs/Web/API/SVGAElement)